### PR TITLE
showmethekey: init at 1.7.3

### PR DESCRIPTION
**Pull request by ocfox**
###### Description of changes
Show keys you typed on screen, so your audiences can see what you do clearly while you are streaming or recording.

This is a [screenkey](https://www.thregr.org/~wavexx/software/screenkey/) alternative, and works not only on X11 but also Wayland.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/showmethekey/default.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  pango,
+  polkit,
+  libinput,
+  gtk3,
+  gtk4,
+  wrapGAppsHook,
+  libxkbcommon,
+  pkg-config,
+  ...
+} @ args:
```
Comment (winterqt) [inline pkgs/applications/video/showmethekey/default.nix:21]:
```suggestion
}:
```

File: pkgs/applications/video/showmethekey/default.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+stdenv.mkDerivation rec {
+  pname = "showmethekey";
+  version = "1.7.3";
+
+  src = fetchFromGitHub {
+    owner = "AlynxZhou";
+    repo = "showmethekey";
+    rev = "v${version}";
+    fetchSubmodules = false;
```
Comment (winterqt) [inline pkgs/applications/video/showmethekey/default.nix:30]: This is the default (please remove the line).

File: pkgs/applications/video/showmethekey/default.nix [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    gtk3
+    gtk4
+  ];
+
+  meta = with lib; {
+    homepage = https://showmethekey.alynx.one/;
+    description = "Show keys you typed on screen";
+    license = licenses.asl20;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.ocfox ];
```
Comment (winterqt) [inline pkgs/applications/video/showmethekey/default.nix:62]:
```suggestion
    maintainers = with maintainers; [ ocfox ];
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -36244,4 +36244,6 @@ with pkgs;
   swift-corelibs-libdispatch = callPackage ../development/libraries/swift-corelibs-libdispatch { };
 
   swaysettings = callPackage ../applications/misc/swaysettings { };
+
+  showmethekey = callPackage ../applications/video/showmethekey { };
```
Comment (winterqt) [inline pkgs/top-level/all-packages.nix:36248]: Please (roughly) alphabetically sort this, don't toss it at the bottom of the file.

**Review by winterqt — CHANGES_REQUESTED** [review]

**Comment by winterqt** [comment]
Also, please swap the order of these commits.

**Comment by ocfox** [comment]
thanks for the review, fixed!

File: pkgs/applications/video/showmethekey/default.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    libevdev
+    libinput
+    libxkbcommon
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    polkit
+    gtk3
+    gtk4
```
Comment (winterqt) [inline pkgs/applications/video/showmethekey/default.nix:52]: Are two versions of GTK necessary? Seems odd.

Comment (ocfox) [reply pkgs/applications/video/showmethekey/default.nix]: Forgot to remove gtk3. sorry

File: pkgs/applications/video/showmethekey/default.nix [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    polkit
+    gtk4
+  ];
+
+  meta = with lib; {
+    homepage = https://showmethekey.alynx.one/;
```
Comment (winterqt) [inline pkgs/applications/video/showmethekey/default.nix:54]:
```suggestion
    homepage = "https://showmethekey.alynx.one/";
```

**Review by winterqt — CHANGES_REQUESTED** [review]
Looks good other than this.

File: pkgs/applications/video/showmethekey/default.nix [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  json-glib,
+  cairo,
+  pango,
+  polkit,
+  libinput,
+  gtk4,
+  wrapGAppsHook,
+  libxkbcommon,
+  pkg-config,
+}:
```
Comment (azahi) [inline pkgs/applications/video/showmethekey/default.nix:19]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, cmake
, glib
, meson
, ninja
, libevdev
, json-glib
, cairo
, pango
, polkit
, libinput
, gtk4
, wrapGAppsHook
, libxkbcommon
, pkg-config
}:
```

File: pkgs/applications/video/showmethekey/default.nix [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  version = "1.7.3";
+
+  src = fetchFromGitHub {
+    owner = "AlynxZhou";
+    repo = "showmethekey";
+    rev = "v${version}";
+    sha256 = "sha256-hq4X4dG25YauMjsNXC6Flco9pEpVj3EM2JiFWbRrPaA=";
+  };
+
+  enableParallelBuilding = true;
```
Comment (azahi) [inline pkgs/applications/video/showmethekey/default.nix:31]: Isn't it the default?

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/showmethekey/default.nix (hunk 7 above)
Comment (ocfox) [reply pkgs/applications/video/showmethekey/default.nix]: Thanks, changed.

File: pkgs/applications/video/showmethekey/default.nix (hunk 8 above)
Comment (ocfox) [reply pkgs/applications/video/showmethekey/default.nix]: removed

**Review by azahi — APPROVED** [review]
LGTM

**Review by winterqt — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/showmethekey/default.nix [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    repo = "showmethekey";
+    rev = "v${version}";
+    sha256 = "sha256-hq4X4dG25YauMjsNXC6Flco9pEpVj3EM2JiFWbRrPaA=";
+  };
+
+  nativeBuildInputs = [
+    glib
+    meson
+    ninja
+    cmake
```
[the remark this answers is not in the document]
Comment (ocfox) [reply pkgs/applications/video/showmethekey/default.nix]: Thanks, also polkit is unnecessary, both removed.

**Review by winterqt — APPROVED** [review]

File: pkgs/applications/video/showmethekey/default.nix [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    glib
+    meson
+    ninja
+    cairo
+    pango
+    json-glib
+    pkg-config
+    libevdev
+    libinput
+    libxkbcommon
```
Comment (bobby285271) [inline pkgs/applications/video/showmethekey/default.nix:38]: Maybe some of these does not belong to nativeBuildInputs? :thinking:

Comment (winterqt) [reply pkgs/applications/video/showmethekey/default.nix:38]: Yup, looks like it. I missed those, sorry :/

Comment (ocfox) [reply pkgs/applications/video/showmethekey/default.nix:38]: glib and pango seems like not. sorry i just keep dependencies write by author but not check its useful. Should i remove it, or wait next version?

Comment (winterqt) [reply pkgs/applications/video/showmethekey/default.nix:38]: What we're saying is that some belong in `buildInputs`. `nativeBuildInputs` is for things like build systems and `pkg-config` -- tools that have to execute during the build; this is to aid cross-compilation. See [this section](https://nixos.org/manual/nixpkgs/unstable/#sec-cross-packaging) of the manual for more information.

Comment (ocfox) [reply pkgs/applications/video/showmethekey/default.nix:38]: So `cairo` and `pango` should belong to `buildInputs`?

Comment (ocfox) [reply pkgs/applications/video/showmethekey/default.nix:38]: So `cairo` and `pango` should belong to `buildInputs`?

**Pull request merged**